src/models/model.py

```
@@ -14,158 +14,251 @@
+
+        # Validate input dimensions with detailed error reporting
         if features != self.input_size:
-            raise ValueError(f"Expected {self.input_size} input features, got {features}")
+            logger.error(
+                f"Input feature mismatch: model expects {self.input_size} features, "
+                f"received {features} features in tensor shape {x.shape}"
+            )
+            raise ValueError(
+                f"Expected {self.input_size} input features, got {features}. "
+                f"Check your data preprocessing and feature selection in config."
+            )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)